x/mint/test/mint_test.go

```diff
@@ -0,0 +1,150 @@
+	s.cctx = cctx
+}
+
+// TestTotalSupplyIncreasesOverTime tests that the total supply of tokens
+// increases over time as new blocks are added to the chain.
+func (s *IntegrationTestSuite) TestTotalSupplyIncreasesOverTime() {
+	require := s.Require()
+
+	initialHeight := int64(1)
+	laterHeight := int64(20)
+
+	err := s.cctx.WaitForNextBlock()
+	require.NoError(err)
+
+	initalSupply := s.getTotalSupply(initialHeight)
+
+	_, err = s.cctx.WaitForHeight(laterHeight)
+	require.NoError(err)
+	laterSupply := s.getTotalSupply(laterHeight)
+
+	require.True(initalSupply.AmountOf(app.BondDenom).LT(laterSupply.AmountOf(app.BondDenom)))
+}
+
+// TestInflationRate verifies that the inflation rate each year matches the
+// expected rate of inflation. See the README.md for the expected rate of
+// inflation.
+func (s *IntegrationTestSuite) TestInflationRate() {
+	require := s.Require()
+
+	type testCase struct {
+		year int
+		want sdktypes.Dec
+	}
+	testCases := []testCase{
+		{year: 0, want: sdktypes.MustNewDecFromStr("8.00")},
+		{year: 1, want: sdktypes.MustNewDecFromStr("7.20")},
+		{year: 2, want: sdktypes.MustNewDecFromStr("6.48")},
+		{year: 3, want: sdktypes.MustNewDecFromStr("5.832")},
+		{year: 4, want: sdktypes.MustNewDecFromStr("5.2488")},
+		{year: 5, want: sdktypes.MustNewDecFromStr("4.72392")},
+		// Note: since the testnode takes time to create blocks, test cases
+		// for years 6+ will time out give the current TimeIotaMs.
+	}
+
+	genesisTime, err := s.cctx.GenesisTime()
+	require.NoError(err)
+
+	lastYear := testCases[len(testCases)-1].year
+	lastTimestamp := genesisTime.Add(time.Duration((lastYear + 1) * minttypes.NanosecondsPerYear))
+	_, err = s.cctx.WaitForTimestampWithTimeout(lastTimestamp, 20*time.Second)
+	require.NoError(err)
+
+	for _, tc := range testCases {
+		startTimestamp := genesisTime.Add(time.Duration(tc.year * minttypes.NanosecondsPerYear))
+		endTimestamp := genesisTime.Add(time.Duration((tc.year + 1) * minttypes.NanosecondsPerYear))
+
+		startHeight, err := s.cctx.HeightForTimestamp(startTimestamp)
+		require.NoError(err)
+		endHeight, err := s.cctx.HeightForTimestamp(endTimestamp)
+		require.NoError(err)
+
+		inflationRate := s.estimateInflationRate(startHeight, endHeight)
+		actualError := inflationRate.Sub(tc.want).Abs()
+		marginOfError := sdktypes.MustNewDecFromStr("0.01")
+		if marginOfError.GT(actualError) {
+			s.Failf("TestInflationRate failure", "inflation rate for year %v is %v, want %v with a .01 margin of error", tc.year, inflationRate, tc.want)
+		}
+	}
+}
+
+func (s *IntegrationTestSuite) getTotalSupply(height int64) sdktypes.Coins {
+	require := s.Require()
+
+	options := &client.ABCIQueryOptions{Height: height}
+	result, err := s.cctx.Client.ABCIQueryWithOptions(
+		context.Background(),
+		"/cosmos.bank.v1beta1.Query/TotalSupply",
+		nil,
+		*options,
+	)
+	require.NoError(err)
+
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
```

Comment:
Do you not have to register the bank codec to be able to unmarshal it?

Also I think if you know the type in the response you can just use proto.Unmarshal

Reply:
result unmarshals correctly without explicitly registering the bank codec but it looks like these two lines can be removed in favor of proto.Unmarshal

Reply:
8e448efe9641150e170e99b52748533cdf338c4f